src/Wig.Zstd/Program.cs

```diff
@@ -29,36 +35,143 @@ public sealed class DefaultCommand : AsyncCommand<DefaultCommand.Settings>
+                }
+            }
+
+            byte[] sourceData = await File.ReadAllBytesAsync(path);
+
+            var compressedData = compressor.Wrap(sourceData);
+
+            if (File.Exists($"{path}.zs") && !overwrite)
+            {
+                AnsiConsole.WriteLine("A compressed file with the same name already exists. Use the -o | --overwrite parameter to force overwrite.");
+                return;
+            }
+
+            await WriteFileAsync(compressedData, path, subfolder, ".zs");
+        }
+
+        public async Task DecompressingAsync(string path, bool overwrite, string subfolder)
+        {
+            using var decompressor = new Decompressor();
+
+            FileAttributes attr = File.GetAttributes(path);
+            if (attr.HasFlag(FileAttributes.Directory))
+            {
+                string[] filePaths = Directory.GetFiles(path);
+                foreach (var filePath in filePaths)
+                {
+                    if (filePath.Contains(".zs"))
+                    {
+                        byte[] compressedData = await File.ReadAllBytesAsync($"{filePath}");
+                        var decompressedData = decompressor.Unwrap(compressedData);
+                        var unpackingPath = Path.ChangeExtension(filePath, "");
+
+                        if (File.Exists($"{path}") && !overwrite)
+                        {
+                            AnsiConsole.WriteLine("A decompressed file with the same name already exists. Use the -o | --overwrite parameter to force overwrite.");
+                            return;
+                        }
+
+                        await WriteFileAsync(decompressedData, unpackingPath, subfolder, "");
+                    }
+                }
+            }
+            if (path.Contains(".zs"))
+            {
+                byte[] compressedData = await File.ReadAllBytesAsync($"{path}");
+                var decompressedData = decompressor.Unwrap(compressedData);
+                var unpackingPath = Path.ChangeExtension(path, "");
+
+                if (File.Exists($"{path}") && !overwrite)
+                {
+                    AnsiConsole.WriteLine("A decompressed file with the same name already exists. Use the -o | --overwrite parameter to force overwrite.");
+                    return;
+                }
+
+                await WriteFileAsync(decompressedData, unpackingPath, subfolder, "");
+            }
+        }
+
         public override async Task<int> ExecuteAsync(CommandContext context, Settings settings)
         {
-            _console.WriteLine("Hello World!");
+            if (settings.IsCompressionMode)
+            {
+                await CompressingAsync(settings.Path, settings.CompressionLevel, settings.Overwrite, settings.Subfolder);
+
+            }
+
+            if (settings.IsDecompressionMode)
+            {
+                await DecompressingAsync(settings.Path, settings.Overwrite, settings.Subfolder);
+            }
 
```

Comment:
can we compress and decompress at the same time? if not, the we need to check to avoid this situation. And think about other parameters validation things